### PR TITLE
fix: Don't propagate DataContext via Border.Child

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): #

Restore the Uno 2.x behaviour whereby Border.ChildProperty doesn't propagate inherited values (including DataContext) to the property value. The inheritance is handled by the visual tree in the normal case (since Child is always a view).

This restores the ability for the framework to call SetParent() to simulate a different inheritance hierarchy than the actual visual tree configuration, eg for AppBarButton using the native style.

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

fixes https://github.com/unoplatform/nventive-private/issues/23

### CODE REVIEW

**Comment by jeromelaban** [comment]
Looks like there are two PRs in one :)

**Comment by davidjohnoliver** [comment]
> Looks like there are two PRs in one :)

Oops I forgot to wait until the other PR was merged 😆 

Should be better now

**Review by carldebilly — APPROVED** [review]

**Review by MatFillion — APPROVED** [review]

**Pull request merged**
